contrib/babelfishpg_tds/src/backend/tds/err_handler.c

```
@@ -356,9 +364,11 @@ emit_tds_log(ErrorData *edata)
 		/* Log the internal error message */
 		ErrorData  *next_edata;
 
+		HOLD_INTERRUPTS();
 		next_edata = CopyErrorData();
 		elog(LOG, "internal error occurred: %s", next_edata->message);
```

we are just ignoring the error here? wouldn't that break the protocol?

---

We are not ignoring the error here, HOLD_INTERRUPTS will just only blocks processing of interrupts.

---

I am not talking about HOLD/RESUME, if send error fails, we are not doing anything in catch other than printing error